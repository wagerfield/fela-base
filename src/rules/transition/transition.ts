import { Unit } from "../../types"
import { QUAD } from "../../core/easing"

export interface TransitionRuleOptions {
  property?: string
  duration?: Unit
  delay?: Unit
  easing?: string
}

export default ({
  property = "all",
  duration = "0.5s",
  delay,
  easing = QUAD.OUT
}: TransitionRuleOptions = {}) => ({
  transitionTimingFunction: easing,
  transitionProperty: property,
  transitionDuration: duration,
  transitionDelay: delay
})
