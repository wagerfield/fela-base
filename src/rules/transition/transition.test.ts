import transition from "./transition"

describe("transition(props)", () => {
  it("renders expected snapshots", () => {
    expect(transition()).toMatchSnapshot()
    expect(
      transition({
        property: "opacity",
        delay: "2s"
      })
    ).toMatchSnapshot()
  })
})
