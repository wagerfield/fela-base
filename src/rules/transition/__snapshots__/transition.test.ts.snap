// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transition(props) renders expected snapshots 1`] = `
Object {
  "transitionDelay": undefined,
  "transitionDuration": "0.5s",
  "transitionProperty": "all",
  "transitionTimingFunction": "cubic-bezier(0.25, 0.46, 0.45, 0.94)",
}
`;

exports[`transition(props) renders expected snapshots 2`] = `
Object {
  "transitionDelay": "2s",
  "transitionDuration": "0.5s",
  "transitionProperty": "opacity",
  "transitionTimingFunction": "cubic-bezier(0.25, 0.46, 0.45, 0.94)",
}
`;
